### DISCUSSION TITLE
Feature request: Additional UI element for selecting one or more items from a list

### DISCUSSION THREAD

**Discussion by andyv09** [MetaMask/snaps#663] [Ideas]
Hey,

We've been wondering on what's your stance on implementing additional UI elements and RPC methods. In our case specifically, a window that looks exactly like the "select the accounts to use on site" (like in the image below), where user is able to pick one or more items. Each item would have a title and perhaps a "view more" button that opens a modal where description of the item is provided.

RPC method invoking this window could accept following parameters; title, description and a list of objects including title and short description. If user accepts a list of selected items is returned.

<img width="1715" alt="Screenshot 2022-07-31 at 20 31 48" src="https://user-images.githubusercontent.com/69682837/182098069-1e700bbe-fd47-4406-b121-a96036257cce.png">

I'm part of BlockchainLab:UM team that is currently working on the [SSI Snap](https://github.com/blockchain-lab-um/ssi-snap). Long story short, we're working on a Snap that is capable of working with DIDs and VCs and would enable MetaMask accounts to become functional DIDs. One of the problems we've stumbled upon is inability to choose a specific VC inside the MetaMask. Currently, this has to be done using a dApp, where a list of all VCs has to be provided to the frontend for user to select one or more. From the perspective of security and data privacy, this is the suboptimal way.

Would something like this make sense? Is this something that could get included in a SIP? Would love to hear your thoughts.

Best regards,
Andy

**Comment by Montoya** [comment]
Hi @andyv09, we have an upcoming feature that will enable building custom UI in MetaMask with Snaps... I will have more info to share within the next couple months about this :)

**Reply by vinny023** [reply]
@Montoya thanks for this answer! Any update on the UI feature or on timing for when we might be able to learn more?

**Reply by andyv09** [reply]
> @Montoya thanks for this answer! Any update on the UI feature or on timing for when we might be able to learn more?

There is a [SIP (Snap improvement proposal)](https://github.com/MetaMask/SIPs/blob/main/SIPS/sip-7.md) regarding UI. You can find. more info about UI there
